tests/phpunit/tests/query/cacheResults.php

```diff
@@ -1211,6 +1211,41 @@ public function data_query_cache_with_empty_result_set() {
 		);
 	}
 
+	/**
+	 * Ensure post meta/term caches are warmed even if a post is already cached.
+	 *
+	 * @covers ::_prime_post_caches
```

Comment:
`@covers WP_Query::get_posts` is included at the class level so I think this will need to go in to a different file to avoid confusing code coverage reports.

It doesn't look like `_prime_post_caches()` has much in the way of tests so that is probably a good thing.

Reply:
Added a bunch of tests in https://github.com/WordPress/wordpress-develop/pull/3658/commits/3c917644b6f3f408d287063e8ed7aa0ca814fba9.